orion/pkg/repos/achieve_repo.go

```diff
@@ -17,22 +17,30 @@ type achieveRepo struct {
 // Interface to implement
 type AchieveRepoInterface interface {
 	Initialize(db *sql.DB)
-	SelectAll() ([]domains.Achieve, error)
-	SelectById(uint) (domains.Achieve, error)
+	SelectAll(bool) ([]domains.Achieve, error)
+	SelectUnpublished() ([]domains.Achieve, error)
 	SelectAllGroupedByYear() ([]domains.AchieveYearGroup, error)
+	SelectById(uint) (domains.Achieve, error)
 	Insert(domains.Achieve) error
 	Update(uint, domains.Achieve) error
 	Delete(uint) error
+	Publish(uint) error
 }
 
 func (ar *achieveRepo) Initialize(db *sql.DB) {
 	ar.db = db
 }
 
-func (ar *achieveRepo) SelectAll() ([]domains.Achieve, error) {
+func (ar *achieveRepo) SelectAll(published bool) ([]domains.Achieve, error) {
```

Comment:
everywhere can you rename variable to `publishedOnly`? So it's clear that setting publishedOnly to true will only grab published rows.